Sensor/toon.py

```diff
@@ -7,19 +7,29 @@
 
 from homeassistant.helpers.entity import Entity
 import custom_components.toon as toon_main
+import datetime as datetime
 
 _LOGGER = logging.getLogger(__name__)
 
+STATE_ATTR_DEVICE_TYPE = "device_type"
+STATE_ATTR_LAST_CONNECTED_CHANGE = "last_connected_change"
+
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Setup sensors."""
+    _toon_main = hass.data[toon_main.TOON_HANDLE]
+
     add_devices([
-        ToonSensor(hass, 'Gas_current', 'CM3'),
-        ToonSensor(hass, 'Gas_today', 'M3'),
         ToonSensor(hass, 'Power_current', 'Watt'),
         ToonSensor(hass, 'Power_today', 'kWh')
     ])
-    _toon_main = hass.data[toon_main.TOON_HANDLE]
+
+    if _toon_main.gas:
```

Comment:
_toon_main references an instance of the ToonDataStore class doesn't it? In there I set the gas and the solar properties to which I'm referring:
self.gas = gas
That makes sense right? I may still be a bit feverish ;-)